### PR TITLE
add support for bitreverse

### PR DESCRIPTION
**Pull request by pranavk**
again, will add tests soon to this PR.

### CODE REVIEW

**Comment by regehr** [comment]
looks great! please add a few tests

**Comment by pranavk** [comment]
okay. tests added

**Comment by pranavk** [comment]
okay. there's one test failure which seems unrelated to me. I will check what's going on

**Comment by pranavk** [comment]
it's definitely related but not obvious to know how my changes affected this test. The problem is that this test (https://github.com/google/souper/blob/master/test/Infer/syn-double-insts/syn-distributive-law-const.opt) fails to infer the RHS after adding support for bitreverse.

**Comment by regehr** [comment]
LGTM, thanks!

**Comment by regehr** [comment]
please resolve conflict and then I'll merge

**Comment by pranavk** [comment]
TODO/note to self: Zhengyang noted that test file contains the hardcoded path of the source file. This needs to be fixed.

**Comment by regehr** [comment]
please don't mix Jenkins changes and bitreverse changes in one PR

**Comment by regehr** [comment]
besides that and resolving the conflict, this is ready to go, I think?

**Comment by pranavk** [comment]
oops. yes. i had no idea i am messing this PR too with my jenkins testing. i will do a force-push and fix it.

Yes, besides the failing test that needs investigation and resolving the conflict, it's ready. i will let you know when it's completely ready to merge

File: unittests/Parser/ParserTests.cpp [hunk 1]
```diff
@@ -608,10 +608,10 @@ cand %2 %3
 %2:i1 = eq 1:i32, %1
 cand %2 1:i1
 )i",
-      R"i(%0:i128 = var ; 0
-%1:i128 = bswap %0
```
Comment (regehr) [inline unittests/Parser/ParserTests.cpp]: wait, why are you deleting an existing test?

Comment (pranavk) [reply unittests/Parser/ParserTests.cpp]: this looks like a typo from the conflicts i resolved from last time; will fix.

**Comment by regehr** [comment]
great, please squash

**Comment by pranavk** [comment]
ok. this is ready to merge now after Zhengyang's fix.

**Comment by regehr** [comment]
thanks!!

**Pull request merged**